junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala

```diff
@@ -0,0 +1,525 @@
+package org.scalajs.junit.plugin
+
+import scala.language.reflectiveCalls
+
+import scala.reflect.internal.Flags
+import scala.tools.nsc._
+import scala.tools.nsc.plugins.{
+  Plugin => NscPlugin, PluginComponent => NscPluginComponent
+}
+
+/* The Scala.js jUnit plugin is a way to overcome make class annotation
+ * information of any test class accessible to the Scala.js testing framework.
+ * This is the information that would usually be accessed through reflection.
+ *
+ *
+ * Test class:
+ *
+ * class Foo {
+ *   @Before def before(): Unit = {
+ *     // Initialize the instance before the tests
+ *   }
+ *   @Test def bar(): Unit = {
+ *     // assert some stuff
+ *   }
+ *   @Ignore("baz not implemented yet") @Test def baz(): Unit = {
+ *     // assert some other stuff
+ *   }
+ * }
+ * object Foo {
+ *   @BeforeClass def beforeClass(): Unit = {
+ *     // Initialize some global state for the tests.
+ *   }
+ * }
```

Comment:
Put this in `{{{` `}}}` for scaladoc. (look at the generated doc if you have doubts)